### PR TITLE
fix(common): justify intervals before hashing

### LINKED ISSUE #3873: distinct on decimal / interval
**Describe the bug**

**To Reproduce**
In PostgreSQL:
```
test=# select distinct * from (values (1.0), (1.00)) as t;
 column1 
---------
     1.0
(1 row)

test=# select distinct * from (values (1.00), (1.0)) as t;
 column1 
---------
    1.00
(1 row)

test=# select distinct * from (values (interval '1' month), (interval '30' day)) as t;
 column1 
---------
 1 mon
(1 row)

test=# select distinct * from (values (interval '30' day), (interval '1' month)) as t;
 column1 
---------
 30 days
(1 row)
```

In RisingWave:
```
dev=> select distinct * from (values (1.0), (1.00)) as t;
   
---
 1
(1 row)

dev=> select distinct * from (values (1.00), (1.0)) as t;
   
---
 1
(1 row)

dev=> select distinct * from (values (interval '1' month), (interval '30' day)) as t;
                  
------------------
 1 mon 00:00:00
 30 days 00:00:00
(2 rows)

dev=> select distinct * from (values (interval '30' day), (interval '1' month)) as t;
                  
------------------
 30 days 00:00:00
 1 mon 00:00:00
(2 rows)
```

**Expected behavior**
Unsure. The PostgreSQL behavior is not preferable either, as it depends on the input order.

**Additional context**

### PR DESCRIPTION
**Pull request by wzzzzd**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

fix https://github.com/singularity-data/risingwave/pull/4575#issuecomment-1211628595

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)

Fixes #3873

### CODE REVIEW

**Comment by xiangjinwu** [comment]
Could you please add the test case to `e2e_test/streaming` to prevent regression?

**Comment by wzzzzd** [comment]
> Could you please add the test case being fixed to `e2e_test/streaming` to prevent regression?



> Could you please add the test case being fixed to `e2e_test/streaming` to prevent regression?

added

**Comment by wzzzzd** [comment]
> Could you please add the test case being fixed to `e2e_test/streaming` to prevent regression?

added

File: e2e_test/streaming/bug_fixes/panic_on_interval_4575_comment.slt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+create table t (v1 interval);
+
+statement ok
+insert into t values (interval '24 h'), (interval '1 d');
+
+statement ok
+create materialized view mv1 as select distinct v1 from t;
+
+statement ok
+drop materialized view mv1;
```
Comment (xiangjinwu) [inline e2e_test/streaming/bug_fixes/panic_on_interval_4575_comment.slt:11]: Let's also select from mv1 to check the result 😂

Comment (wzzzzd) [reply e2e_test/streaming/bug_fixes/panic_on_interval_4575_comment.slt]: added

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**